### DISCUSSION TITLE
How to include a DataTemplate from a separate file to UserControl.DataTemplates collection?

### DISCUSSION THREAD

**Discussion by ppetrov-dev** [AvaloniaUI/Avalonia#7602] [Q&A] [answered]
Hi there!
I have a lot of different data types with own logic and view and I want to extract DataTemplate to a separate file for each types and use it in a UserControl's ListBox, how to implement it? I don't want to create a view for each types. Is it possible?

**The examples from docs** (https://docs.avaloniaui.net/docs/templates/data-templates)
![image](https://user-images.githubusercontent.com/11585634/153870853-30f5aa2f-b484-4121-99b7-e7abfb8a5a02.png)

**What I expect:**
```
<Window.DataTemplates>
    <ResourceInclude Source='StudentDataTemplate.axaml' />
    <ResourceInclude Source='TeacherDataTemplate.axaml' />
</Window.DataTemplates>
```

**Comment by CollinAlpert** [comment]
I actually figured this out while reading through the [Resources docs](https://docs.avaloniaui.net/docs/styling/resources).
Put your DataTemplates inside `ResourceDictionaries` (one per file), give the templates a key, reference the resource dictionaries in your main XAML file and then reference the `DataTemplates` there using `StaticResource`.

```xml
<ResourceDictionary xmlns="https://github.com/avaloniaui"
                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
    <DataTemplate DataType="MyViewModel"
                  x:Key="MyTemplate">
<!--Content here-->
    </DataTemplate>
</ResourceDictionary>
```

```xml
<Window.Resources>
    <ResourceDictionary>
        <ResourceDictionary.MergedDictionaries>
            <ResourceInclude Source="/Views/MyView.axaml"></ResourceInclude>
        </ResourceDictionary.MergedDictionaries>
    </ResourceDictionary>
</Window.Resources>

<ContentControl Content="{Binding MyViewModel}" ContentTemplate="{StaticResource MyTemplate}" />
```

**Reply by jinek** [reply]
In original question data templates don't have a key and can be selected by type.

**Comment by hheimerd** [comment, marked as answer]
Found it. Use StaticResource  tag
``` axaml
<TabControl.Resources>
      <ResourceDictionary>
          <ResourceDictionary.MergedDictionaries>
              <ResourceInclude Source="/Resources/Templates/Template.axaml" />
          </ResourceDictionary.MergedDictionaries>
      </ResourceDictionary>
  </TabControl.Resources>
  <TabControl.DataTemplates >
      <StaticResource ResourceKey="TemplateKey"></StaticResource>
  </TabControl.DataTemplates>
<TabControl.Resources>
```

**Reply by jinek** [reply]
You have introduced key and now template can't be selected automatically based on data type.
